org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java

```diff
@@ -92,24 +82,73 @@ public String toString() {
 		return version;
 	}
 
+	/**
+	 * Tries to read the Xtext version from this bundle's manifest. The method opens the
+	 * manifest of org.eclipse.xtext.util and reads the <code>Maven-Version</code> value from
+	 * it. In CI scenarios this will be "unspecified", then it continues to read the
+	 * <code>Bundle-Version</code> value. If this ends with ".qualifier", it replaces the qualifier
+	 * by "-SNAPSHOT". Otherwise it will return the base version without the qualifier.
+	 * 
+	 * @return Version string in format <code>MAJOR.MINOR.MICRO[-SNAPSHOT]</code>
+	 */
 	private static String readVersionFromManifest() {
-		URL baseURL = XtextVersion.Plugin.INSTANCE.getBaseURL();
-		try (InputStream is = new URL (baseURL + "META-INF/MANIFEST.MF").openStream()) {
-			Manifest manifest = new Manifest(is);
-			String version = manifest.getMainAttributes().getValue("Maven-Version");
-			if ("unspecified".equals(version)) {
-				version = manifest.getMainAttributes().getValue("Bundle-Version");
-				if (version.endsWith(".qualifier")) {
-					return version.replace(".qualifier", "-SNAPSHOT");
-				} else {
-					return version.substring(0, version.lastIndexOf("."));
+		try {
+			URL ownManifestURL = getOwnManifestURL();
+			
+			try (InputStream is = ownManifestURL.openStream()) {
+				Manifest manifest = new Manifest(is);
+				String version = manifest.getMainAttributes().getValue("Maven-Version");
+				if ("unspecified".equals(version)) {
+					version = manifest.getMainAttributes().getValue("Bundle-Version");
+					if (version.endsWith(".qualifier")) {
+						return version.replace(".qualifier", "-SNAPSHOT");
+					} else {
+						return version.substring(0, version.lastIndexOf("."));
+					}
 				}
+				return version;
 			}
-			return version;
 		} catch (IOException e) {
 			throw new RuntimeException(e);
 		}
 	}
+	
+	/**
+	 * Locates the URL of this bundle's manifest file. This is done by locating the manifest file
+	 * relative to the class resource of this class. The location can be either be a file URL in the context
+	 * of unit test execution or an archive URL for CI/release bundle.
+	 * 
+	 * @throws FileNotFoundException When the MANIFEST.MF for org.eclipse.xtext.util was not found
+	 */
+	private static URL getOwnManifestURL () throws FileNotFoundException {
```

Comment:
I'd prefer something along the lines of 
```
ClassLoader cl = getClass().getClassLoader();
Enumeration<URL> allManifestMF = cl.getResources(""META-INF/MANIFEST.MF";
while(allManifestMf.hasMoreElements()) {
  // check the URL for xtext.util and proceed
  
}
```

Reply:
I had this approach first, too, but it did not work. I did not came across the manifest of org.eclipse.xtext.util. What was I missing?

Reply:
My assumption is that Gradle knows nothing about that 
As the manifest is not on the classpath when not executed with a jar but from eclipse